src/steps/do-query/do-get.js

```diff
@@ -63,20 +64,56 @@ export default function(requestContext, responseContext, registry) {
 
 }
 
-function parseFields(fieldsParam) {
+function parseFields(fieldsParam, registry) {
   let fields;
   if(typeof fieldsParam === "object") {
     fields = {};
     let isField = (it) => !arrayContains(["id", "type"], it);
 
     for(let type in fieldsParam) {
-      let provided = parseCommaSeparatedParam(fieldsParam[type]) || [];
+      let provided = parseCommaSeparatedParam(fieldsParam[type], type, registry) || [];
       fields[type] = provided.filter(isField);
     }
   }
   return fields;
 }
 
-function parseCommaSeparatedParam(it) {
-  return it ? it.split(",").map(decodeURIComponent) : undefined;
+function parseCommaSeparatedParam(it, type, registry) {
+  return it ? it.split(",").map(f => {
+    return transformDotSeparatedPath(f, type, registry);
+  }) : undefined;
+}
+
+function transformDotSeparatedPath(it, type, registry) {
+  let parts = it.split(".");
+  let types = [type];
+  parts.forEach((p, index) => {
+    if (index < parts.length - 1) {
+      types.push(registry.dbAdapter(types[index]).constructor.getReferencedType(types[index], p));
+    }
+  });
+
+  return parts.map((part, index) => {
+    if (shouldCamelizeType(types[index], registry)) {
+      return camelize(decodeURIComponent(part));
+    }
+    return decodeURIComponent(part);
+  }).join(".");
+}
+
+function transformFilters(it, type, registry) {
```

Comment:
Hmm. It occurs to me now that *all* the camelizing and dasherizing needs to traverse down into objects... filters may need this in a case like

```js
// ?filter[simple][$or][first-name]=joe&filter[simple][$or][first-name]=jim
...where({
  $or: [{ firstName: 'joe' }, { firstName: 'jim' }]
})
```
Crap, so it needs to traverse into arrays too? This is getting weird.

Additionally, complex JSON values are allowed by the spec. So, you could get this back:

```json
{
  "data": {
    "id": "1",
    "attributes": {
      "location": {
        "street-address": "1 Pacific Ave"
      }
    }
  }
}
```

I guess I will start making this recursive. Unless you just want to draw the line here and say “filters are serialized and passed to the adapter as they are with no transformations.” But, I think it would still be nice to deep-dasherize complex attributes. What do you think?